### PR TITLE
SIW v2 OIE - terminal view

### PR DESCRIPTION
**Pull request by haishengwu-okta**
- The first version of terminal state.
- Clean up `AppState` - remove unused properties
- https://okta.box.com/s/z01ooxgzbyxm85f6xlwp4tl7iga4wpqg

### CODE REVIEW

File: src/v2/models/AppState.js [hunk 1]
```diff
@@ -46,9 +40,41 @@ export default Model.extend({
...
+    let vo;
+    if (!_.isEmpty(this.get('remediation'))) {
+      vo = this.get('remediation').filter(r => r.name === currentFormName)[0];
+    }
+
+    if (!vo) {
+      if (currentFormName) {
+        Logger.warn(`Cannot find view state for form ${currentFormName}. Fall back to terminal state.`);
+      }
+      vo = this.get('currentState').terminal[0];
```
Comment (nikhilvenkatraman-okta) [inline src/v2/models/AppState.js]: Can we add some comments here for our assumptions that when we don't have a remediation, we are in a terminal state ?

File: src/v2/BaseLoginRouter.js [hunk 2]
```diff
@@ -42,13 +42,13 @@ function beaconIsAvailable (Beacon, settings) {
   return true;
 }
 
-function loadLanguage (appState, i18n, assetBaseUrl, assetRewrite) {
+function loadLanguage (appState, languageCode, i18n, assetBaseUrl, assetRewrite) {
```
Comment (nikhilvenkatraman-okta) [inline src/v2/BaseLoginRouter.js]: what is this change for ?

File: src/v2/view-builder/internals/BaseView.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ export default View.extend({
     // Create Model
     const IonModel = this.createModelClass();
     const model = new IonModel ({
-      formName: this.options.remediationValue.name,
+      formName: this.options.currentViewState.name,
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/internals/BaseView.js]: Maybe we should just call this `this.options.currentViewState.formName` to be clear?

Comment (nikhilvenkatraman-okta) [reply src/v2/view-builder/internals/BaseView.js]: ah, nvm i think this is what the api sends us right ? formName comes under the `name` property

File: src/v2/models/AppState.js [hunk 4]
```diff
@@ -46,9 +40,41 @@ export default Model.extend({
     },
   },
 
-  parse: function (options) {
-    this.settings = options.settings;
-    return _.extend(_.omit(options, 'settings'), { languageCode: this.settings.get('languageCode') });
+  getCurrentViewState () {
+    const currentFormName = this.get('currentFormName');
+    let vo;
```
Comment (nikhilvenkatraman-okta) [inline src/v2/models/AppState.js]: can we rename this variable ?

File: src/v2/BaseLoginRouter.js (hunk 2 above)
Comment (haishengwu-okta) [reply src/v2/BaseLoginRouter.js]: I remove `languageCode` from `AppState` hence has to get it from `settings`.

File: src/v2/view-builder/internals/BaseView.js (hunk 3 above)
Comment (haishengwu-okta) [reply src/v2/view-builder/internals/BaseView.js]: `currentViewState` is from `remediation` hence it is `name'`

**Pull request merged**